test/tests.py

```diff
@@ -1607,12 +1612,14 @@ def analyze_c_files(self, files, func_spans, cblock_spans, not_linemarked):
                     if curr_cblock_expected:
                         (expected_dml_lineno,
                          expected_c_lineno) = curr_cblock_expected.pop()
-                        assert ((dml_lineno, c_lineno)
-                                == (expected_dml_lineno, expected_c_lineno)), \
-                               (f'In {curr_file}: Expected DML line '
-                                + f'{expected_dml_lineno} at '
-                                + f'{expected_c_lineno}, not DML line '
-                                + f'{dml_lineno} at {c_lineno}')
+                        if ((dml_lineno, c_lineno)
+                            != (expected_dml_lineno, expected_c_lineno)):
+                            raise self.Error(
+                                [(curr_file, expected_c_lineno,
+                                  f"Wrong DML linenumber {dml_lineno}"),
+                                 (Path(self.filename).resolve(),
+                                  expected_dml_lineno,
+                                  f"Expected line {expected_dml_lineno}")])
```

Comment:
What makes a proper `TestFailure` that much better than the simpler `assert` to the point that you bothered to do this?

Reply:
Also, recommend touching up the language a bit:
`f"Redirected to wrong DML line"`
and then
`f"Expected redirection to this line, was redirected to {dml_lineno} instead"`

Reply:
> What makes a proper `TestFailure` that much better than the simpler `assert` to the point that you bothered to do this?

only for my own convenience, it made them clickable in emacs

Reply:
Language is not a priority for error messages in tests, but doesn't hurt so I'll grab your formulations as-is.